src/slc/mailrouter/browser/views.py

```diff
@@ -1,120 +1,137 @@
 import email
+import six
 import sys
-from tempfile import NamedTemporaryFile
 import traceback
-from StringIO import StringIO
+from io import TextIOWrapper
+from logging import getLogger
+from tempfile import NamedTemporaryFile
 
 from Acquisition import aq_inner
-from zope.component import queryUtility, getAllUtilitiesRegisteredFor
-from zope.interface import alsoProvides
+from plone.uuid.interfaces import IUUID
+from Products.CMFCore.interfaces import IFolderish
+from Products.CMFCore.permissions import AddPortalContent
+from Products.CMFCore.utils import _checkPermission
+from Products.CMFPlone.PloneBatch import Batch
 from Products.Five import BrowserView
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
-from Products.CMFPlone.PloneBatch import Batch
 from Products.statusmessages.interfaces import IStatusMessage
-from Products.CMFCore.interfaces import IFolderish
-from Products.CMFCore.utils import _checkPermission
-from Products.CMFCore.permissions import AddPortalContent
-from plone.uuid.interfaces import IUUID
-from slc.mailrouter.interfaces import IFriendlyNameStorage
-from slc.mailrouter.interfaces import IMailRouter
-from slc.mailrouter.exceptions import PermanentError
-from slc.mailrouter.exceptions import PermissionError
-from slc.mailrouter.exceptions import TemporaryError
+from six import StringIO
 from slc.mailrouter import MessageFactory as _
+from slc.mailrouter.exceptions import PermanentError, PermissionError, TemporaryError
+from slc.mailrouter.interfaces import IFriendlyNameStorage, IMailRouter
 from slc.mailrouter.utils import store_name
+from zope.component import getAllUtilitiesRegisteredFor, queryUtility
+from zope.interface import alsoProvides
+
+try:
+    from Products.CMFPlone.utils import safe_nativestring
+except ImportError:
```

Comment:
If we are going to support only 5.2 this BBB code can go

Reply:
True! Removing.